hw.py

```diff
@@ -0,0 +1,241 @@
+# Реализуйте CRUD операции для каждой из таблиц через создание маршрутов, REST API (итого 15 маршрутов).
+# * Чтение всех
+# * Чтение одного
+# * Запись
+# * Изменение
+# * Удаление
+
+# Данная промежуточная аттестация оценивается по системе "зачет" / "не зачет"
+
+# "Зачет" ставится, если Слушатель успешно выполнил задание.
+# "Незачет" ставится, если Слушатель не выполнил задание.
+
+# Критерии оценивания:
+# 1 - Слушатель создал базу данных для интернет-магазина. База данных должна состоять из трёх таблиц: товары, заказы и пользователи.
+# — Таблица «Товары» должна содержать информацию о доступных товарах, их описаниях и ценах.
+# — Таблица «Заказы» должна содержать информацию о заказах, сделанных пользователями.
+# — Таблица «Пользователи» должна содержать информацию о зарегистрированных пользователях магазина.
+
+
+from typing import List
+from fastapi import FastAPI
+from pydantic import BaseModel, Field
+import databases
+import sqlalchemy
+import datetime
+
+
+DATABASE_URL = 'sqlite:///mydatabase.db'
+
+database = databases.Database(DATABASE_URL)
+metadata = sqlalchemy.MetaData()
+
+#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[SQLAlchemy Tables]~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+users = sqlalchemy.Table(
+    'users',
+    metadata,
+    sqlalchemy.Column('id', sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column('first_name', sqlalchemy.String(15)),
+    sqlalchemy.Column('last_name', sqlalchemy.String(20)),
+    sqlalchemy.Column('email', sqlalchemy.String(50), unique=True),
+    sqlalchemy.Column('password', sqlalchemy.String(20))
+)
+
+
+products = sqlalchemy.Table(
+    'products',
+    metadata,
+    sqlalchemy.Column('id', sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column('name', sqlalchemy.String(50)),
+    sqlalchemy.Column('description', sqlalchemy.String(1000)),
+    sqlalchemy.Column('cost', sqlalchemy.Float),
+)
+
+
+orders = sqlalchemy.Table(
+    'orders',
+    metadata,
+    sqlalchemy.Column('id', sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column('user_id', sqlalchemy.Integer, sqlalchemy.ForeignKey('users.id')),
+    sqlalchemy.Column('product_id', sqlalchemy.Integer, sqlalchemy.ForeignKey('products.id')),
+    sqlalchemy.Column('order_date', sqlalchemy.Date),
+    sqlalchemy.Column('order_status', sqlalchemy.String(20))
+)
+#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[Pydantic Models]~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+class UserIn(BaseModel):
+    first_name: str = Field(..., max_length=15)
+    last_name: str = Field(..., max_length=20)
+    email: str = Field(..., max_length=50)
+    password: str = Field(..., max_length=20)
+
+
+class User(BaseModel):
+    id: int
+    first_name: str = Field(..., max_length=15)
+    last_name: str = Field(..., max_length=20)
+    email: str = Field(..., max_length=50)
+    password: str = Field(..., max_length=20)
```

Comment:
Эти классы очень похожи. Значит, мы можем создать один, наследуясь от другого.
```suggestion
class UserIn(BaseModel):
    first_name: str = Field(..., max_length=15)
    last_name: str = Field(..., max_length=20)
    email: str = Field(..., max_length=50)
    password: str = Field(..., max_length=20)


class User(UserIn):
    id: int
```